public_html/wp-content/themes/pattern-directory/src/components/pattern-preview/index.js

```diff
@@ -14,21 +14,68 @@ import DragHandle from './drag-handle';
 
 /* eslint-disable jsx-a11y/anchor-is-valid -- These are just placeholders. */
 
-const INITIAL_WIDTH = 1200;
+const INITIAL_WIDTH = 960;
 
 function PatternPreview( { blockContent } ) {
+	const showViewportControl = useViewportMatch( 'mobile', '>=' );
+	const showViewportControlDefault = useViewportMatch( 'large', '>=' );
+	const showViewportControlLarge = useViewportMatch( 'wide', '>=' );
+
 	const instanceId = useInstanceId( PatternPreview );
 	const [ width, setWidth ] = useState( window.innerWidth < INITIAL_WIDTH ? window.innerWidth : INITIAL_WIDTH );
-	const onDragChange = useCallback(
-		( delta ) => {
-			setWidth( ( value ) => value + delta );
-		},
-		[ setWidth ]
-	);
+	const onDragChange = useCallback( ( delta ) => setWidth( ( value ) => value + delta ), [ setWidth ] );
+
+	const availableWidths = useMemo( () => {
+		// Less than 480 wide.
+		if ( ! showViewportControl ) {
+			return [];
+		}
+		if ( showViewportControlLarge ) {
+			// More than 1280 wide.
+			return [
+				{ label: 'Full (1200px)', value: 1200 },
+				{ label: 'Default (960px)', value: 960 },
+				{ label: 'Medium (480px)', value: 480 },
+				{ label: 'Narrow (320px)', value: 320 },
+			];
+		} else if ( showViewportControlDefault ) {
+			// Less than 1280, more than 960.
+			return [
+				{ label: 'Default (960px)', value: 960 },
+				{ label: 'Medium (480px)', value: 480 },
+				{ label: 'Narrow (320px)', value: 320 },
+			];
+		}
+		// Less than 960, but larger than 480.
+		return [
+			{ label: __( 'Medium (480px)', 'wporg-patterns' ), value: 480 },
+			{ label: __( 'Narrow (320px)', 'wporg-patterns' ), value: 320 },
+		];
+	}, [ showViewportControl, showViewportControlDefault, showViewportControlLarge ] );
+
+	let currentOpt = false;
+	if ( -1 === availableWidths.map( ( opt ) => opt.value ).indexOf( width ) ) {
```

Comment:
Although inconsequential for this application, if you are just trying to see if it exists `.some` will break on `true`, saving a few `map` iterations.

Reply:
Ah, good point, that just seems more straightforward too. ✔️